src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsClientModelValidatorProvider.cs

```diff
@@ -5,26 +5,32 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Linq;
+using Microsoft.AspNet.Mvc.DataAnnotations;
 using Microsoft.Extensions.Localization;
 using Microsoft.Extensions.OptionsModel;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
 {
     /// <summary>
+    /// A factory for validators based on ValidationAttribute.
+    /// </summary>
+    /// <param name="attribute"></param>
+    /// <param name="stringLocalizer"></param>
+    /// <returns></returns>
+    public delegate IAttributeAdapter ValidationAttributeAdapterFactory(
+        ValidationAttribute attribute,
+        IStringLocalizer stringLocalizer);
+
+    /// <summary>
     /// An implementation of <see cref="IClientModelValidatorProvider"/> which provides client validators
     /// for attributes which derive from <see cref="ValidationAttribute"/>. It also provides
     /// a validator for types which implement <see cref="IClientModelValidator"/>.
     /// The logic to support <see cref="IClientModelValidator"/>
-    /// is implemented in <see cref="DataAnnotationsClientModelValidator{}"/>.
+    /// is implemented in <see cref="ValidationAttributeAdapter{}"/>.
     /// </summary>
     public class DataAnnotationsClientModelValidatorProvider : IClientModelValidatorProvider
     {
-        // A factory for validators based on ValidationAttribute.
-        internal delegate IClientModelValidator DataAnnotationsClientModelValidationFactory(
-            ValidationAttribute attribute,
-            IStringLocalizer stringLocalizer);
-
-        private readonly Dictionary<Type, DataAnnotationsClientModelValidationFactory> _attributeFactories =
+        private readonly Dictionary<Type, ValidationAttributeAdapterFactory> _attributeFactories =
             BuildAttributeFactoriesDictionary();
```

Comment:
Don't need this field. Just use `ValidationAttributeAdapterTable.AttributeFactories` directly.